app/src/main/java/de/danoeh/antennapod/menuhandler/FeedItemMenuHandler.java

```diff
@@ -254,15 +255,44 @@ public static void removeNewFlagWithUndo(@NonNull Fragment fragment, FeedItem it
             }
         };
 
+        int playStateStringRes;
+        switch (playState) {
+            default:
+            case FeedItem.UNPLAYED:
+                if (item.getPlayState() == FeedItem.NEW) {
+                    //was new
+                    playStateStringRes = R.string.removed_new_flag_label;
+                } else {
+                    //was played
+                    playStateStringRes = R.string.marked_unread_label;
+                }
+                break;
+            case FeedItem.PLAYED:
+                playStateStringRes = R.string.marked_read_label;
+                break;
+        }
+
+        int duration = Snackbar.LENGTH_LONG;
+
+        if (showSnackbar) {
+            ((MainActivity) fragment.getActivity()).showSnackbarAbovePlayer(
+                    playStateStringRes, duration)
+                    .setAction(fragment.getString(R.string.undo), v -> {
+                        DBWriter.markItemPlayed(item.getPlayState(), item.getId());
+                        // don't forget to cancel the thing that's going to remove the media
+                        h.removeCallbacks(r);
+                    });
+        }
+
+        h.postDelayed(r, (int) Math.ceil(duration * 1.05f));
+    }
+
+    public static void removeNewFlagWithUndo(@NonNull Fragment fragment, FeedItem item) {
+        markReadWithUndo(fragment, item, FeedItem.UNPLAYED, false);
+    }
 
-        Snackbar snackbar = ((MainActivity) fragment.getActivity()).showSnackbarAbovePlayer(
-                R.string.removed_new_flag_label, Snackbar.LENGTH_LONG)
-                .setAction(fragment.getString(R.string.undo), v -> {
-                    DBWriter.markItemPlayed(FeedItem.NEW, item.getId());
-                    // don't forget to cancel the thing that's going to remove the media
-                    h.removeCallbacks(r);
-                });
-        h.postDelayed(r, (int) Math.ceil(snackbar.getDuration() * 1.05f));
+    public static void addToQueue(Context context, FeedItem item) {
+        DBWriter.addQueueItem(context, item);
```

Comment:
Please just call the method directly instead of adding a wrapper that just executes a single line